experimental/report/renderer.go

```diff
@@ -0,0 +1,893 @@
+			cur.underlines = append(cur.underlines, strings.TrimRight(sidebar+string(buf), " "))
+
+			// Then, splat in the message. having two rows like this ensures that
+			// each message has one pipe directly above it.
+			if idx >= 0 {
+				ul := rest[idx]
+
+				actualStart := ul.start - 1
+				for _, other := range rest[idx:] {
+					if other.start <= ul.start {
+						actualStart += len(c.BoldForLevel(ul.level))
+					}
+				}
+				for len(buf) < actualStart+len(ul.message)+1 {
+					buf = append(buf, ' ')
+				}
+
+				// Make sure we don't crop *part* of an escape. To do this, we look for
+				// the last ESC in the region we're going to replace. If it is not
+				// followed by an m, we need to insert that many spaces into buf to avoid
+				// overwriting it.
+				writeTo := buf[actualStart:][:len(ul.message)]
+				lastEsc := bytes.LastIndexByte(writeTo, 033)
+				if lastEsc != -1 && !bytes.ContainsRune(writeTo[lastEsc:], 'm') {
+					// If we got here, it means we're going to crop an escape if
+					// we don't do something about it.
+					spaceNeeded := len(writeTo) - lastEsc
+					for i := 0; i < spaceNeeded; i++ {
+						buf = append(buf, 0)
+					}
+					copy(buf[actualStart+lastEsc+spaceNeeded:], buf[actualStart+lastEsc:])
+				}
+
+				copy(buf[actualStart:], ul.message)
+			}
+			cur.underlines = append(cur.underlines, strings.TrimRight(sidebar+string(buf), " "))
+		}
+
+		return true
+	})
+
+	//nolint:dupword
+	// Now that we've laid out the underlines, we can add the starts and ends of all
+	// of the multilines, which go after the underlines.
+	//
+	// The result is that a multiline will look like this:
+	//
+	//   code
+	//  ____^
+	// | code code code
+	// \______________^ message
+	var line strings.Builder
+	for i := range info {
+		cur := &info[i]
+		prevStart := -1
+		for j, ml := range cur.sidebar {
+			if ml == nil {
+				continue
+			}
+
+			line.Reset()
+			var isStart bool
+			switch w.start + i {
+			case ml.start:
+				if ml.startWidth == 0 {
+					continue
+				}
+
+				isStart = true
+				fallthrough
+			case ml.end:
+				// We need to be flush with the sidebar here, so we trim the trailing space.
+				sidebar := []byte(strings.TrimRight(renderSidebar(0, -1, prevStart, c, cur.sidebar[:j+1]), " "))
+
+				// We also need to erase the bars of any multis that are before this multi
+				// and start/end on the same line.
+				if !isStart {
+					for i, otherML := range cur.sidebar[:j+1] {
+						if otherML != nil && otherML.end == ml.end {
+							// We assume all the color codes have the same byte length.
+							codeLen := len(c.bBlue)
+							idx := i*(2+codeLen) + codeLen
+							if idx < len(sidebar) {
+								sidebar[idx] = ' '
+							}
+						}
+					}
+				}
+
+				// Delete the last pipe and replace it with a slash or space, depending.
+				// on orientation.
+				line.Write(sidebar[:len(sidebar)-1])
+				if isStart {
+					line.WriteByte(' ')
+				} else {
+					line.WriteByte('\\')
+				}
```

Comment:
Two reasons:
1. If I did I would feel compelled to port my https://docs.rs/boxy/latest/boxy to Go, and this is enough work as is.
2. I... don't like how they look, aesthetically? I've used them for diagnostics before and they're pretty meh imo.

The thing with the slashes here is new (to me). I think it looks nicer, personally. Can't really use `+` because we use underscores to be able to achieve effects like `\____-` for multiline note spans, for example.